documentation/Docker-DNS.md

```diff
@@ -0,0 +1,86 @@
+# Docker DNS setup
+
+There are several DNS components in the Docker setup:
+
+* A non-validating resolver, used for DNS resolving by almost all tests.
+  As we have our own DNSSEC validation test, we want to see bogus responses as well.
+* A validating resolver, used to validate DANE records through ldns-dane.
+* An authoritative server for the connection test zone.
+* The DNS records in the zone for `INTERNETNL_DOMAINNAME`.
+  These are hosted elsewhere, but have certain requirements for the instance to work.
+
+The resolvers do not require any specific configuration.
+In this document, `INTERNETNL_DOMAINNAME` is `example.com`.
+
+## Common parts
+
+For accessing the absolute minimum basic functionality of the application the following DNS records must be configured:
+
+    example.com.                    A      192.0.2.1
+                                   AAAA   2001:db8:1::1
+
+The following extra records must be configured for language switching:
+
+    www.example.com.                CNAME  example.com.
+    nl.example.com.                 CNAME  example.com.
+    en.example.com.                 CNAME  example.com.
+
+The hostname (`INTERNETNL_DOMAINNAME`) should have SPF, DKIM and DMARC,
+as some mail servers may filter on this, and it could affect mail tests.
+For a domain that does not otherwise send email, use:
+
+    example.com.		           TXT	"v=spf1 a -all"	; The "a" mechanism is needed for the mail test (see rfc7208, section-2.3).
+    _domainkey.example.com.	       TXT	"v=DKIM1; p="	; empty DKIM to score 100% for this non-sending subdomain that does have SPF "a" mechanism which is needed for mail test.
+    _dmarc.example.com.	           TXT	"v=DMARC1; p=reject; sp=reject;"
+
+The `INTERNETNL_DOMAINNAME` host should also have a working MX and correct FCrDNS.
+
+
+## Specific settings for batch mode
+
+For batch, the connection test is not used, and the authoritative server should not be publicly available.
+Set `IPV4_IP_PUBLIC=127.0.0.1` and `IPV6_IP_PUBLIC=::1` in `docker/host.env`.
+
+
+## Specific settings for single test mode
+
+The single test instance also has an IPv6-only host:
+
+    ipv6.example.com.               AAAA   2001:db8:1::1
```

Comment:
Maybe it's indeed best to be complete here, but maybe adding a comment that of course normally this would be used:
```zone
$ORIGIN example.com.
```
Since then almost all `.example.com.` suffixes could be removed.

Reply:
BTW great that the official example IP addresses are used!

Reply:
I just did not want to leave any chance for misconfiguration :)